gnovm/pkg/gnolang/debugger.go

```diff
@@ -0,0 +1,528 @@
+package gnolang
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// DebugState is the state of the machine debugger.
+type DebugState int
+
+const (
+	DebugAtInit DebugState = iota
+	DebugAtCmd
+	DebugAtRun
+	DebugAtExit
+)
+
+// Debugger describes a machine debugger state.
+type Debugger struct {
```

Comment:
Can you add small unit test for Debugger so that developers can check how to use Debugger?